common/src/main/java/org/jboss/hal/testsuite/util/AvailablePortFinder.java

```diff
@@ -0,0 +1,173 @@
+package org.jboss.hal.testsuite.util;
+
+import java.io.IOException;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Finds currently available server ports.
+ *
+ * @author <a href="http://mina.apache.org">Apache MINA Project</a>
+ * @see <a href="http://www.iana.org/assignments/port-numbers">IANA.org</a>
+ */
+public class AvailablePortFinder {
+    /**
+     * The minimum number of user port number.
+     */
+    public static final int MIN_PORT_NUMBER = 1024;
+
+    /**
+     * The maximum number of user port number.
+     */
+    public static final int MAX_PORT_NUMBER = 49151;
+
+    /**
+     * Creates a new instance.
+     */
+    private AvailablePortFinder() {
```

Comment:
The javadoc is incorrect. This constructor actually prevents creating instance of this class.